srml/elections-phragmen/src/lib.rs

```diff
@@ -159,6 +161,11 @@ decl_storage! {
 		/// The present candidate list. Sorted based on account id. A current member can never enter
 		/// this vector and is always implicitly assumed to be a candidate.
 		pub Candidates get(fn candidates): Vec<T::AccountId>;
+
+		/// Has the storage format been updated?
+		/// NOTE: Only use and set to false if you have used an early version of this module. Should
+		/// be set to true otherwise.
+		DidMigrate: bool = false;
```

Comment:
should just initialise this to `true` by here? also will it better to use a `u8` version byte instead?
upgrade existing chain won't reinitialise this and will have `DidMigrate` to be `None` which defaults to `false`

Reply:
Good ideas.

Reply:
no need for version byte. once the upgrade is through, a second update should remove the item from storage, and then the storage item declaration may be removed from code completely.

this process can be repeated whenever necessary and doesn't leave an additional set of storage items around indefinitely.

Reply:
can be left without default, or as is with = false.

in general we'd want an additional genesis item that set it to true, but setting a default isn't the way to do that.